include/nuttx/wdog.h

```diff
@@ -88,13 +88,8 @@ struct wdog_s
   FAR void          *picbase;    /* PIC base address */
 #endif
   clock_t            expired;    /* Timer associated with the absoulute time */
-};
-
-struct wdog_period_s
-{
-  struct wdog_s      wdog;       /* Watchdog */
   clock_t            period;     /* Period time in ticks */
```

Comment:
but the chagne will increase wdog_s size even the caller doesn't require the period callback.

Reply:
how about add a kconfig setting to make period wdog feature configurable?

Reply:
doesn't resolve the problem, since not all user in the same image use the periodic watchdog.